lib/auth/middleware.go

```diff
@@ -391,23 +403,66 @@ func (a *Middleware) ValidateClientVersion(ctx context.Context, info IdentityInf
 	clientVersion, err := semver.NewVersion(clientVersionString)
 	if err != nil {
 		logger.WithError(err).Warn("Failed to determine client version")
+		a.displayRejectedClientAlert(ctx)
 		if err := info.Conn.Close(); err != nil {
 			logger.WithError(err).Warn("Failed to close client connection")
 		}
+
 		return trace.AccessDenied("client version is unsupported")
 	}
 
 	if clientVersion.LessThan(*a.OldestSupportedVersion) {
 		logger.Info("Terminating connection of client using unsupported version")
+		a.displayRejectedClientAlert(ctx)
+
 		if err := info.Conn.Close(); err != nil {
 			logger.WithError(err).Warn("Failed to close client connection")
 		}
+
 		return trace.AccessDenied("client version is unsupported")
 	}
 
 	return nil
 }
 
+// displayRejectedClientAlert creates an alert to notify admins that
+// unsupported Teleport versions exist in the cluster and are explicitly
+// being denied to prevent causing issues. Alerts are limited to being
+// created once per day to reduce backend writes if there are a large
+// number of unsupported clients constantly being rejected.
+func (a *Middleware) displayRejectedClientAlert(ctx context.Context) {
+	if a.AlertCreator == nil {
+		return
+	}
+
+	now := time.Now()
+	lastAlert := a.lastRejectedAlertTime.Load()
+	if !now.Add(24 * time.Hour).After(time.Unix(lastAlert, 0)) {
+		return
+	}
+
+	if !a.lastRejectedAlertTime.CompareAndSwap(lastAlert, now.UnixNano()) {
+		return
+	}
+
+	alert, err := types.NewClusterAlert(
+		"rejected-unsupported-connection",
+		fmt.Sprintf("Connections were rejected from agents running unsupported Teleport versions(<%s), they will be inaccessible until upgraded.", a.OldestSupportedVersion),
+		types.WithAlertSeverity(types.AlertSeverity_MEDIUM),
+		types.WithAlertLabel(types.AlertOnLogin, "yes"),
+		types.WithAlertLabel(types.AlertVerbPermit, "token:create"),
+	)
+	if err != nil {
+		log.WithError(err).Warn("failed to create unsupposted client alert")
```

Comment:
s/unsupposted/unsupported/ - although I think `unsupported-client` might be better as it can sound like it failed to create a client alert because it was unsupported, rather than it being an alert about unsupported clients. Same with the error a few lines down except the spelling there is correct.